great_expectations/data_context/types/base.py

```diff
@@ -2339,20 +2369,43 @@ def __deepcopy__(self, memo):
         result = cls.__new__(cls)
 
         memo[id(self)] = result
-        for key, value in self.to_dict().items():
-            value_copy = safe_deep_copy(data=value, memo=memo)
-            setattr(result, key, value_copy)
+
+        attributes_to_copy = set(CheckpointConfigSchema().fields.keys())
+        for key in attributes_to_copy:
+            try:
+                value = self[key]
+                value_copy = safe_deep_copy(data=value, memo=memo)
+                setattr(result, key, value_copy)
+            except AttributeError as e:
+                pass
```

Comment:
I don't think we ever want a raw `pass` if possible. Could we `logger.debug` this? It won't ever get picked up unless a dev requests it.

Reply:
@cdkini Let us discuss what makes sense to do here (if anything).  Thanks.

Reply:
Warning added.